src/DataSourceProvider.vue

```diff
@@ -217,12 +236,10 @@ export default {
         .then(dataSource => {
           this.selectedDataSource = dataSource;
 
-          Fliplet.Widget.emit('dataSourceSelect',
-            {
-              columns: this.selectedDataSource.columns,
-              id: this.selectedDataSource.id
-            }
-          );
+          Fliplet.Widget.emit('dataSourceSelect', {
```

Comment:
Is there a reason we can't just send the entire object? Rather than tweaking what attributes we send, the component caller can make use of all the available data.

```suggestion
          Fliplet.Widget.emit('dataSourceSelect', this.selectedDataSource);
```

Reply:
@tonytlwu There is no reason for us not to do so.
I'll make a new commit with this change.